cocos/animation/exotic-animation/exotic-animation.ts

```diff
@@ -680,6 +701,86 @@ interface ExoticTrackEvaluationRecord<TValue> {
     evaluator: ExoticTrackEvaluator<TValue>;
 }
 
+/**
+ * Exotic TRS animation graph evaluator.
+ */
+export class ExoticTrsAGEvaluation {
+    constructor (nodeAnimations: ExoticNodeAnimation[], context: AnimationClipGraphBindingContext) {
+        this._nodeEvaluations = nodeAnimations.map(
+            (nodeAnimation) => nodeAnimation.createEvaluatorForAnimationGraph(context),
+        ).filter((x) => !!x) as ExoticNodeAnimationEvaluatorX[];
+    }
+
+    public destroy () {
+        this._nodeEvaluations.forEach((nodeEvaluation) => {
+            nodeEvaluation.destroy();
+        });
+    }
+
+    public evaluate (time: number, pose: Pose) {
+        this._nodeEvaluations.forEach((nodeEvaluator) => {
+            nodeEvaluator.evaluate(time, pose);
+        });
+    }
+
+    private _nodeEvaluations: ExoticNodeAnimationEvaluatorX[];
+}
+
+class ExoticNodeAnimationEvaluatorX {
```

Comment:
Why is there an X at the end of the name?

Reply:
https://github.com/cocos/cocos-engine/pull/13503/commits/2843818640584bc64db480c4889c4815bc20e505